src/app/components/chat/message.ts

```diff
@@ -1,13 +1,14 @@
 import { ChatUser } from './user';
 
-export type ChatMessageType = 0 | 1;
+export type ChatMessageType = 'normal' | 'system' | 'invite';
 
 export const CHAT_MESSAGE_MAX_CONTENT_LENGTH = 1000;
 export const TIMEOUT_CONSIDER_QUEUED = 1500; // Time in ms until a queued message should be displayed as such.
 
 export class ChatMessage {
-	static readonly TypeNormal = 0;
-	static readonly TypeSystem = 1;
+	static readonly TypeNormal = 'normal';
+	static readonly TypeSystem = 'system';
+	static readonly TypeInvite = 'invite'; 
```

Comment:
I think what we should do here is use a const enum instead of these declarations:

Declare the const enum above the ChatMessage class like this:
```
export const enum ChatMessageType {
  NORMAL = 'normal',
  INVITE = 'invite',
}
```

In the ChatMessage class, the type definition for `type` stays the same:
```
type!: ChatMessageType;
```

Then, you can remove the `static readonly Type*` fields.

In ChatClient (line 426), you can then use:
```
type: ChatMessageType.NORMAL,
```
instead of 
```
type: ChatMessage.TypeNormal,
```

And for when you want to use the const enum in Vue view components:
In the TS file, import the const enum type into the local context:

```
readonly ChatMessageType = ChatMessageType;
```

And then you can use it like that in a Vue file:
```
		:class="{
			'chat-msg-type-normal': message.type === ChatMessageType.NORMAL,
```

In addition to all that, I don't think type "system" is in use, so you can remove it.